xla/service/gpu/tests/gemm_rewrite_test.cc

```diff
@@ -5771,66 +5771,41 @@ TEST_P(ParameterizedFp8GemmRewriteTest, ScaledABUnscaledDWithAllGatherF8) {
+      y_unscaled = f32[16,32] multiply(y_f32, y_scale_bcast)
+      all_gather = f32[16,64]{1,0} all-gather(x_unscaled), channel_id=1, replica_groups={{0,1},{2,3},{4,5},{6,7}}, dimensions={1}, use_global_device_ids=true
+      all_gather1 = f32[64,32]{1,0} all-gather(y_unscaled), channel_id=2, replica_groups={{0,2,4,6},{1,3,5,7}}, dimensions={0}, use_global_device_ids=true
+      ROOT dot_a = f32[16,32] dot(all_gather, all_gather1), lhs_contracting_dims={1}, rhs_contracting_dims={0}
           }
 )";
 
-  RunAndFilecheckHloRewrite(hlo_text,
-                            GemmRewriter(se::CudaComputeCapability{
-                                se::CudaComputeCapability::HOPPER, 0}),
-                            R"(
-; CHECK-LABEL: ENTRY %test (x: f8e4m3fn[16,16], y: f8e4m3fn[32,16], b: f16[16], x_scale: f16[], y_scale: f16[]) -> f16[16,16] {
-; CHECK:         [[P0:%[^ ]+]] = f8e4m3fn[16,16]{1,0} parameter(0)
-; CHECK:         [[P3:%[^ ]+]] = f16[] parameter(3)
-; CHECK:         [[AG:%[^ ]+]] = f8e4m3fn[16,32]{1,0} all-gather([[P0]]), {{[^ ]+}}
-; CHECK:         [[P1:%[^ ]+]] = f8e4m3fn[32,16]{1,0} parameter(1)
-; CHECK:         [[P1_TRANSPOSE:%[^ ]+]] = f8e4m3fn[16,32]{1,0} transpose([[P1]]), dimensions={1,0}
-; CHECK:         [[CV:%[^ ]+]] = f32[] convert([[P3]])
-; CHECK:         [[P4:%[^ ]+]] = f16[] parameter(4)
-; CHECK:         [[CV1:%[^ ]+]] = f32[] convert([[P4]])
-; CHECK:         [[C:%[^ ]+]] = f32[] constant(1)
-; CHECK:         [[B:%[^ ]+]] = f16[16]{0} parameter(2)
-; CHECK:         ROOT [[GEMM:%[^ ]+]] = f16[16,16]{1,0} custom-call([[AG]], [[P1_TRANSPOSE]], [[CV]], [[CV1]], [[C]], /*index=5*/[[C]], [[B]]),
-; CHECK:           custom_call_target="__cublas$lt$matmul$f8",
-; CHECK:           backend_config={
-; CHECK-DAG:         "alpha_real":1
-; CHECK-DAG:         "alpha_imag":0
-; CHECK-DAG:         "beta":0
-; CHECK-DAG:         "dot_dimension_numbers":{
-; CHECK-DAG:           "lhs_contracting_dimensions":["1"]
-; CHECK-DAG:           "rhs_contracting_dimensions":["1"]
-; CHECK-DAG:           "lhs_batch_dimensions":[]
-; CHECK-DAG:           "rhs_batch_dimensions":[]
-; CHECK-DAG:         }
-; CHECK-DAG:         "precision_config":{
-; CHECK-DAG:           "operand_precision":["DEFAULT","DEFAULT"]
-; CHECK-DAG:         }
-; CHECK-DAG:         "epilogue":"BIAS"
-; CHECK:           }
-      )");
+  HloModuleConfig config = GetModuleConfigForTest();
+  config.set_use_spmd_partitioning(true);
+  config.set_num_partitions(8);
+  TF_ASSERT_OK_AND_ASSIGN(auto module,
+                          ParseAndReturnVerifiedModule(hlo_text, config));
+  GemmRewriter pass(
+      se::CudaComputeCapability{se::CudaComputeCapability::HOPPER, 0});
+  TF_ASSERT_OK_AND_ASSIGN(bool changed, this->RunHloPass(&pass, module.get()));
+  EXPECT_TRUE(changed);
+
+  EXPECT_THAT(
+      module->entry_computation()->root_instruction(),
+      GmockMatch(
+          m::CustomCall({"__cublas$lt$matmul$f8"}).WithShape(F32, {16, 32})));
```

Comment:
Why not test the full HLO is correct? That way you can check the all-gather is still there and has its dtype changed to fp8

Reply:
Resolved.